app/utils/build/__init__.py

```diff
@@ -230,106 +237,99 @@ def parse_build_data(build_data, job, kernel, errors, build_dir=None):
     :type errors: dictionary
     :param build_dir: Full path to the build directory.
     :type build_dir: string
-    :return A 2-tuple: A BuildDocument or None, a dictionary with artifacts
-    keys and names.
+    :return A 2-tuple: (BuildDocument, a "artifact type": "artifact name" dictionary)
     """
-    build_doc = None
-    artifacts = {}
+    if not isinstance(build_data, types.DictionaryType):
+        raise BuildError(500, "JSON data is not a dictionary")
+
+    data = build_data
+
+    try:
+        defconfig = data[models.DEFCONFIG_KEY]
+        d_job = data.get(models.JOB_KEY, job)
```

Comment:
Saving a method bound to an object reduced the code readability. It's easier to understand the line:
```
job = data.get("job")
```
compared to your example. Trying to figure out what the above line does, you only need to know what the object is - it's in the function header, and looks like a dictionary.
To know what `data_get` does, the reader must first backtrack to find where it was defined before guessing/finding out what it actually does.

The code doesn't seem performance-critical, so I replaced this with the readable versoin.